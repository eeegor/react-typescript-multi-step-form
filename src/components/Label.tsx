/**
 * Component Label
 */

import * as React from 'react';
import * as classnames from 'classnames';
import './Label.scss';

export interface Props {
	className?: string;
	htmlFor?: string;
	label: string | JSX.Element | undefined;
}

export class Label extends React.Component<Props> {
	render(): JSX.Element | null {
		const { className, label, htmlFor } = this.props;

		if (!label || label === '') {
			return null;
		}

		return (
			<label className={classnames('label', className)} htmlFor={htmlFor}>
				{label}
			</label>
		);
	}
}
