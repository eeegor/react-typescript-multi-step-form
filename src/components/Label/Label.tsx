/**
 * Component Label
 */

import React from 'react';
import classnames from 'classnames';
import './Label.scss';

export interface Props {
	className?: string;
	htmlFor?: string;
	label: string | JSX.Element | undefined;
}

export const Label = (props: Props): JSX.Element | null => {
	const { className, label, htmlFor } = props;

	if (!label || label === '') {
		return null;
	}

	return (
		<label className={classnames('label', className)} htmlFor={htmlFor}>
			{label}
		</label>
	);
};
