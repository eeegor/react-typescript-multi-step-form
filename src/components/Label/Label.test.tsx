import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Label } from './Label';
import { mount } from 'enzyme';

describe('Label', () => {
	it('renders markup correctly', () => {
		const wrapper = <Label label="hello" />;
		const tree = renderer.create(wrapper).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should not output a label with empty value', () => {
		const wrapper = mount(<Label label="" />);
		expect(wrapper.find('.label')).toHaveLength(0);
	});
});
