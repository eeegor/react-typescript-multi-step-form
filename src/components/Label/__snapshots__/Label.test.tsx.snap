// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders markup correctly 1`] = `
<label
  className="label"
>
  hello
</label>
`;
